File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
class Rectangle:
    """
    class Rectangle with two private attributes, width and height
    """
    def __init__(self, width=0, height=0):
        self.height = height
        self.width = width

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        if type(value) != int:
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        if type(value) is not int:
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    def area(self):
        return self.__height * self.__width

    def perimeter(self):
        if self.__width == 0 or self.__height == 0:
            return ""
        else:
            return (self.__height + self.__width) * 2

    def __str__(self):
        new_str = ""
        if self.__width == 0 or self.__height == 0:
            return new_str
        else:
            for j in range(self.__height):
                for k in range(self.__width):
                    new_str += '#'
                new_str += '\n'
            new_str = new_str.rstrip()
            return new_str

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node

class Rectangle {
  constructor (w, h) {
    if (w < 0 || h < 0) {
    } else if (w > 0 && h > 0) {
      this.width = w;
      this.height = h;
    }
  }

  // method
  print () {
    let i = 0;
    const num = 'X';
    for (i; i < this.height; i++) {
      console.log(num.repeat(this.width));
    }
  }
}
module.exports = Rectangle;
